File: clients/js/src/features/resolve_ens_text_record.ts
import {
  ClientConfigBuilder,
  PolywrapClient,
  Uri
} from "@polywrap/client-js";

export async function runTestCase(input: unknown): Promise<void> {
  if (typeof input !== "string" || !Uri.isValidUri(input)) {
    throw Error(
      "resolve_ens_text_record test case input must be a string & valid URI"
    );
  }

  const uri = Uri.from(input);

  console.log(`URI Authority: ${uri.authority}`);

  const config = new ClientConfigBuilder()
    .addDefaults()
    .build();

  const client = new PolywrapClient(config);

  console.log(`Resolving: ${uri}`);

  const result = await client.tryResolveUri({ uri });

  if (result.ok) {
    console.log(`Received: ${result.value.type}`);
    console.log("Success!");
  }
}

File: clients/rs/src/features/resolve_ens_text_record.rs
use std::error::Error;
use polywrap_client::{client::PolywrapClient, builder::PolywrapClientConfig, core::{uri_resolver_handler::UriResolverHandler, resolution::uri_resolution_context::UriPackageOrWrapper, uri::Uri}};
use serde_json::Value;

pub fn run_test_case(input: &Value) -> Result<(), Box<dyn Error>> {
  let uri: Uri = serde_json::from_value::<String>(input.clone())?.try_into()?;
  let uri_authority = uri.authority();
  let uri_string = uri.to_string();

  println!("URI Authority: {uri_authority}");

  let config = PolywrapClientConfig::new();
  let client: PolywrapClient = PolywrapClient::new(config.into());

  println!("Resolving: {uri_string}");

  let result = client.try_resolve_uri(&uri.try_into()?, None)?;
  let result_type = match result {
    UriPackageOrWrapper::Uri(_) => "uri",
    UriPackageOrWrapper::Wrapper(_, _) => "wrapper",
    UriPackageOrWrapper::Package(_, _) => "package",
  };

  if let UriPackageOrWrapper::Uri(_) = result {
    println!("Received: '{result_type}'");
    println!("Success!");
  }

  Ok(())
}